api/lib/domain/usecases/link-user-to-organization-student-data.js

```diff
@@ -0,0 +1,34 @@
+const { NotFoundError, UserNotAuthorizedToAccessEntity } = require('../../domain/errors');
+
+module.exports = async function linkUserToOrganizationStudentData({
+  code,
+  user,
+  campaignRepository,
+  studentRepository,
+}) {
+  let student;
+  const campaign = await campaignRepository.getByCode(code);
+  const organizationId = campaign.organizationId;
+
+  if (user.id === null) {
+    throw new UserNotAuthorizedToAccessEntity('User is not part of the organization student list');
+  }
+
+  const students = await studentRepository.findByOrganizationIdAndUserInformation({
+    organizationId,
+    firstName: user.firstName,
+    lastName: user.lastName,
+    birthdate: user.birthdate,
+  });
+
+  if (students.length === 1) {
+    student = students[0];
+  } else {
+    throw new NotFoundError('Not found only 1 student');
+  }
+
+  const patchedStudent = await studentRepository.associateUserAndStudent({ userId: user.id, studentId: student.id });
```

Comment:
Ne devrait on pas gérer le cas où le student a déjà un `userId` avant son association ? Ici cela ne pose pas de problème s'il est déjà rattaché, est-ce ok ?

Reply:
En effet, d'ailleurs à ce sujet je pense ajouter une clause à l'association coté repo: `where userId is not null`
Comme ça on ne pourrais pas re-associer un student. Tu en penses quoi ?

Reply:
ok, et du coup, ça throw une erreur ?

Reply:
Non c'est juste que le repo ne renverrais pas un student qui à déjà était réconcilié (qui à déjà un userId).
Mais en effet je pense que se système serait intéressant avec l'ajout de la route `GET user/{id}/student` que vous pourriez appeler en front avant d'afficher la page de réconciliation.